src/odemis/driver/rigol.py

```diff
@@ -248,67 +248,67 @@ def terminate(self):
             self.socket.close()
         self._is_connected = False
 
-    def sendOrderCommand(self, cmd, val=""):
+    def sendOrderCommand(self, cmd, val=b""):
         """
         Sends one command, and don't expect any reply
-        cmd (str): command to send
-        val (str): value to send (if any)
+        cmd (byte str): command to send
+        val (byte str): value to send (if any)
         raises:
             IOError: if problem with sending/receiving data over the connection
         """
         if not self._is_connected:
             raise IOError("Device %s not connected." % (self._host,))
 
-        msg = "%s %s\n" % (cmd, val)
+        msg = b"%s %s\n" % (cmd, val)
 
         with self._net_access:
             self.socket.sendall(msg)
 
     def sendQueryCommand(self, cmd):
         """
         Sends one command, and expect a reply
-        cmd (str): command to send, including the ?
+        cmd (byte str): command to send, including the ?
```

Comment:
I think it'd keep the code easier to read if cmd (and val) would be just "str" (basestring on Python2). It would then be converted to bytes by `sendQueryCommand()` and `sendOrderCommand()` (using ascii). The response seems to always be "sane" (ie, no weird character), so it should also be fine to decode it back immediately into a string. 

BTW, could you update the docstring here to indicate the type of the return value.

Reply:
done